sips/sip-02x/sip-02x-non-sequential-multisig-transactions.md

```diff
@@ -0,0 +1,152 @@
+# Preamble
+
+SIP Number: 02x
+
+Title: Non-sequential Multisig Transactions
+
+Authors: Jeff Bencin <jbencin@hiro.so>, Vlad Bespalov <vlad@asigna.io>
+
+Consideration: Technical
+
+Type: Consensus
+
+Status: Draft
+
+Created: 2023-08-30
+
+License: BSD 2-Clause
+
+Sign-off:
+
+Discussions-To: https://github.com/stacksgov/sips
+
+# Abstract
+
+This SIP proposes a new multisig transaction format which is intended to be easier to use than the current format described in SIP-005.
+It does not remove support for the current format, rather it is intended to co-exist with the old format and give users a choice of which format to use.
+
+The issue with the current format is that it establishes a signer order when funds are sent to multisig account address, and requires signers to sign in the same order to spend the funds.
+In practice, the current format has proven proven difficult for developers to understand and implement, as evidenced by the lack of Stacks multisig implementations today.
+
+This new format intends to simplify the signing algorithm and remove the requirement for in-order signing, without comprimising on security or increasing transaction size.
+It is expected that this will lead to better wallet support for Stacks multisig transactions.
+
+# Introduction
+
+Currently, a multisig transaction requires the first signer to sign the transaction itself, and following signers to sign the signature of the previous signer.
+For a transaction with *n* signers, the final signature is generated in the following way:
+
+```
+signature_n(...(signature_2(signature_1(tx))))
+```
+
+There are a few drawbacks to doing it this way:
+
+- The order in which the signers must sign is fixed as soon as funds are send to a multisig account, which limits flexibility when creating a spending transaction from a multisig account
+- The process of signing a transaction requires each signer to validate the entire signature chain before signing, in order to make sure it matches the transaction, leading to `O(n^2)` signing times
+- This does not reduce the size of a transaction, as each intermediate signature must still be included
+- The algorithm for doing this is complex, and several developers have a hard time understanding and implementing it correctly
+
+This document proposes having each signer sign the transaction directly:
+
+```
+signature_1(tx), signature_2(tx), ..., signature_n(tx)
+```
+
+This would address all of the concerns listed above, and would not increase transaction size or make it easier to forge a signature
+
+## Examples
+
+Imagine a DAO that has a management team comprised of five members.
+They create a 3 out of 5 multisig account on Stacks.
```

Comment:
Combine lines 60-61:
```
Imagine a DAO that has a management team comprised of five members, and tey create a 3 out of 5 multisig account on Stacks.
```

Reply:
In general it's considered a best practice in Markdown to write one sentence per line. It displays the same, but makes diffs and merges easier